core/src/main/java/com/linecorp/armeria/server/file/AbstractHttpFile.java

```diff
@@ -197,7 +197,15 @@ public final CompletableFuture<HttpResponse> read(Executor fileReadExecutor, Byt
         requireNonNull(alloc, "alloc");
 
         return readAttributes(fileReadExecutor)
-                .thenApply(attrs -> read(fileReadExecutor, alloc, attrs))
+                .thenApply(attrs -> {
+                    final HttpResponse res = read(fileReadExecutor, alloc, attrs);
+                    if (res != null) {
+                        return res;
+                    }
+
+                    // read() returned null above.
```

Comment:
I think we can remove this comment because it's obvious.
How about adding a log instead?
```suggestion
                    logger.debug("{} is not found. attrs: {}", pathOrUri(), attrs);
```
I think we can also add the same log here: https://github.com/line/armeria/blob/3b35c0573f40bd8f69e0b4895b6af2efd75e6908/core/src/main/java/com/linecorp/armeria/server/file/AbstractHttpFile.java#L263